### ISSUE TITLE
AntRunner does not use ProjectHelperRegistry

### ISSUE THREAD

**Issue by jgsuess** [eclipse-platform/eclipse.platform#205]
Ant is meant to allow polyglot interpretation using [custom Project Helpers](https://ant.apache.org/manual/projecthelper.html).

The build file below should prompt ant to consult the Project Helper Repository
```xml
<?xml version="1.0" encoding="UTF-8"?>
<project>
	<import file="custom.format" />
</project>
```

Eclipse's `InternalAntRunner` does not do that. Instead it selects the [current project helper](https://github.com/eclipse-platform/eclipse.platform/blob/master/ant/org.eclipse.ant.core/src_ant/org/eclipse/ant/internal/core/ant/InternalAntRunner.java#L390), which is [the first project helper found in the classpath](https://ant.apache.org/manual/api/org/apache/tools/ant/ProjectHelper.html#getProjectHelper()).

This behaviour is incorrect. Instead, it should call [`ProjectHelper.configureProject()`](https://github.com/apache/ant/blob/master/src/main/org/apache/tools/ant/ProjectHelper.java#L100) first. This correct behaviour is implemented by all `ant` invocations in the ant distribution, as visible from the call graph below.

![image](https://user-images.githubusercontent.com/782737/194190448-f3a98a74-19a7-413d-9b1d-0e4f8191a023.png)

This a snippet of the implementation of the function.

```java
    /**
     * Configures the project with the contents of the specified build file.
     *
     * @param project The project to configure. Must not be <code>null</code>.
     * @param buildFile A build file giving the project's configuration.
     *                  Must not be <code>null</code>.
     *
     * @exception BuildException if the configuration is invalid or cannot be read
     */
    public static void configureProject(Project project, File buildFile) throws BuildException {
        FileResource resource = new FileResource(buildFile);
        ProjectHelper helper = ProjectHelperRepository.getInstance().getProjectHelperForBuildFile(resource);
        project.addReference(PROJECTHELPER_REFERENCE, helper);
        helper.parse(project, buildFile);
    }
    ```

Could you please add this, as it blocks polyglot use of ant.

**Comment by jukzi** [comment]
please contribute a PR along with a junit test,

**Comment by akurtakov** [comment]
I would like to expand on @jukzi reply here: In the past Eclipse was huge on Ant integration and practically using it for everything build related on our side. Lately, most things are moved (or at least being moved) to Maven and Ant integration is in deep maintenance mode so enhancements like you ask for need someone (you?) to step up and do them.

**Comment by jgsuess** [comment]
I agree. I'm happy to help out here. I think the major issue committees to get started with the setup of the tools. We are using armed to produce an integrated tool chain for MDE. See [publication here](https://www.researchgate.net/project/BotBuilder-8-Open-MDE-DevOps-Toolchain).

**Comment by akurtakov** [comment]
There is https://wiki.eclipse.org/Eclipse_Platform_SDK_Provisioning which is supposed to provide a full-featured ready to use SDK for people wanting to contribute patches.
